examples/hello-keycloak/src/router.js

```diff
@@ -24,8 +24,17 @@ const router = new Router({
   ]
 })
 
-router.beforeEach((to, from, next) => {
+function sleep(ms) {
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
+
+router.beforeEach(async (to, from, next) => {
   if (to.matched.some(record => record.meta.requiresAuth)) {
+    // We wait for Keycloak init, then we can call all methods safely
+    while (!router.app.$keycloak.ready) {
```

Comment:
As you say, not a very pretty solution. But works 😄  

Would probably be good with a kind of limit on this though, In case `router.app.$keycloak.ready` is never ready

Reply:
Yes, good point! Maybe we can put a "timeout" on `sleep` function.